src/main/java/dev/ebullient/convert/tools/dnd5e/Tools5eJsonSourceCopier.java

```diff
@@ -355,7 +351,7 @@ private JsonNode resolveDynamicText(String originKey, JsonNode value, JsonNode t
             return value;
         }
         Matcher matcher = variable_subst.matcher(value.toString());
-        while (matcher.find()) {
+        if (matcher.find()) {
```

Comment:
why did you make this change? matcher.find() iterates over all matches, there can be more than one.

Reply:
Because there's an unguarded `return` in the loop body, it will only ever execute once (or never, if there are no matches). In which case `if` is clearer because it makes it clear that it only ever executes once. If that's not intentional then there's a pre-existing bug here.